Firestore/Source/API/FIRFirestoreSettings.mm

```diff
@@ -35,6 +35,9 @@
 static const int64_t kMinimumCacheSizeBytes = 1 * 1024 * 1024;
 static const BOOL kDefaultTimestampsInSnapshotsEnabled = YES;
 
+// Public constant
+extern "C" const int64_t kFIRFirestoreCacheSizeUnlimited = api::Settings::CacheSizeUnlimited;
```

Comment:
When I did the api::Settings port I was thinking about moving all the defaults to be defined there but wasn't sure if that would open the door for static initialization order fiasco, so I left it as-is.  But I guess this is safe because it's a builtin value.

Reply:
The key to making that work is to ensure that the values are compiler initialized. Making values `constexpr` is one guaranteed way to do that. Another is to annotate the value with `ABSL_CONST_INIT` which adds an attribute that would cause a compiler error if the value could not be compiler initialized.

With that in place it's safe to refer to other values in the definition of a constant. I've ported all the settings defaults as well.